src/cpp/session/modules/SessionFind.cpp

```diff
@@ -39,14 +39,75 @@ using namespace rstudio::core;
 
 namespace rstudio {
 namespace session {
-namespace modules {   
+namespace modules {
 namespace find {
 
 namespace {
 
 // This must be the same as MAX_COUNT in FindOutputPane.java
 const size_t MAX_COUNT = 1000;
 
+// Reflects the estimated current progress made in performing a replace
+class LocalProgress : public boost::noncopyable
+{
+public:
+
+   explicit LocalProgress(int max, int updateFrequency)
+   {
+      units_ = 0;
+      max_ = max;
+      updateFrequency_ = updateFrequency;
+      updateIncrement_ = static_cast<int>((updateFrequency_ * max_) / 100);
+      if (updateIncrement_ < 1)
+         updateIncrement_ = 1;
+      nextUpdate_ = updateIncrement_;
+   }
+
+   void addUnit()
+   {
+      units_++;
+      if (units_ >= nextUpdate_)
+      {
+         nextUpdate_ += updateIncrement_;
+         if (nextUpdate_ > max_)
+            nextUpdate_ = max_;
+         notifyClient();
+      }
+   }
+
+   void addUnits(int num)
+   {
+      units_ += num;
+      while (units_ >= nextUpdate_)
+      {
+         nextUpdate_ += updateIncrement_;
+         if (nextUpdate_ > max_)
+            nextUpdate_ = max_;
+         notifyClient();
+         // prevent infinite loop when max is reached
+         if (units_ >= max_)
+            break;
+      }
+   }
+
+private:
+
+   void notifyClient()
+   {
+      json::Object data;
+      data["max"] = max_;
+      data["units"] = units_;
+      module_context::enqueClientEvent(
+            ClientEvent(client_events::kReplaceUpdated, data));
+   }
+
+   int units_;
+   int max_;
```

Comment:
Max represents the total number of units expected to be replaced. This is used for the ReplacedUpdated message which is a small message to let the frontend know what progress is being made.

Reply:
In that case could this be named something a little more descriptive, like `completedReplaceCount_`? Maybe that's too long, but something along those lines.

Reply:
Oops I meant that response for the `units_` conversation. Maybe both members could be renamed something like `replacedCount_` and `totalReplaceCount_`? Then their relationship would be clear from the names, I think.

Also, if `max_` isn't supposed to change after object construction, it can be made `const`.

Reply:
Yeah I like these names better, I've made the updates